server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,230 @@
+/**
+* The /fee_centers HTTP API endpoint
+*
+* @module finance/feeCenters
+*
+* @description This controller is responsible for implementing all crud and others custom request
+* on the fee_center table through the `/fee_centers` endpoint.
+*
+* @requires lib/db
+**/
+
+
+var db = require('../../lib/db');
+var NotFound = require('../../lib/errors/NotFound');
+const FilterParser = require('../../lib/filter');
+
+/**
+* Returns an array of fee centers
+*
+* @example
+* // GET /fee_centers : Get list of fee_centers
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.list(req, res, next);
+*/
+
+function list (req, res, next) {
+  'use strict';
+
+  let ordering = 'ORDER BY fc.label';
+  let is_detailed = Boolean(Number(req.query.detailed));
+  let detailed = is_detailed ? ', fc.note, fc.is_principal, fc.project_id, fc.is_cost, p.name, p.abbr, p.enterprise_id, p.zs_id ' : '';
+
+  let sql =
+    `SELECT fc.id, fc.label, fc.is_cost ${detailed} FROM fee_center AS fc JOIN project as p on fc.project_id = p.id`;
+
+  let notInStatement = `(fc.id NOT IN (SELECT s.cc_id FROM service AS s WHERE NOT ISNULL(s.cc_id) UNION SELECT s.pc_id FROM service AS s WHERE NOT ISNULL(s.pc_id)) AND ?)`;
+
+  let filterParser = new FilterParser(req.query, { tableAlias : 'fc' });
+
+  filterParser.equals('is_principal');
+  filterParser.equals('is_cost');
+  filterParser.custom('available', notInStatement, req.query.available);
+  filterParser.setOrder(ordering);
+
+  let query = filterParser.applyQuery(sql);
+  let params = filterParser.parameters();
+
+  db.exec(query, params)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* Create a fee center in the database
+*
+* @example
+* // POST /fee_centers : Insert a fee center
+* var feeCenters = require('finance/fee_centers');
```

Comment:
Giving example code in your comments is an excellent idea.  It makes the usage clear, and it can potentially be extracted and run by a documentation program to make sure that the function behaves as expected.

However, this code will not work, because the file name is `finance/feeCenter.js`.  Could you either remove the misleading comment or update the name?   This problem has been raised in another review https://github.com/IMA-WorldHealth/bhima-2.X/pull/1105#discussion_r98898911.